### PR TITLE
ENH: Adequate module tests to ITK test driver structure.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Use the itkTestDriver to set up the required environment for the test
suite to be run under the ITK testing framework.

Make the necessary modifications to the CMakeLists.txt file.

Rename the test file main method name to adapt to these changes.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@rdebroiz  Review and test the patch set please.

**Comment by rdebroiz** [comment]
I got this: `undefined reference to 'itkTBBImageToImageFilterTest(int, char**)'`

File: test/itkTBBImageToImageFilterTest.cxx [hunk 1]
```diff
@@ -73,7 +73,7 @@ class TBBImageToImageFilterHelper : public TBBImageToImageFilter< TInputImage, T
 
 } // itk
 
-int main(int argc, const char** argv)
+int itkTBBImageToImageFilterTest(int argc, const char** argv)
```
Comment (rdebroiz) [inline test/itkTBBImageToImageFilterTest.cxx]: You should replace `(int argc, const char** argv)` by `(int , char* [])` and then remove the `(void)argc; (void)argv;`

Comment (rdebroiz) [reply test/itkTBBImageToImageFilterTest.cxx]: Done in:  59957fa

**Comment by rdebroiz** [comment]
I am not sure to have the expected behaviour, when i launch CTest, it wait for a user input to select a test to launch (wait for the input 0). There is no indication to let you know it, unless you use the verbose option `-V`

**Comment by jhlegarreta** [comment]
Sorry, my mistake, apart from the test name for the `itkTestDriver`, its argument (main method/filename) was missing. Please, check that it works now.

**Pull request merged**